src/app/services/downloader.service.ts

```diff
@@ -69,8 +69,23 @@ export class DownloaderService {
       await new Promise<void>((resolve, reject) => {
         new ReadableStream({
           start(controller) {
-            const reader = response.body.getReader();
-            read();
+            let reader = null;
+            if(downloadObject.data.isDataset) {
+              response.json().then(json => {
+                reader = json.records.filter(function(record){return record.length > 0}).map(function(record){return record.trim()}).join("\n");
+                const blob = new Blob([reader], { type: 'text/plain' });
+                createAndDownloadElement(blob, downloadObject.data.path);
+                resolve();
+              })
+              .catch(error => {
+                context.log.severe("An error occurred downloading " + fileName + " : ", error)
+                controller.error(error);
```

Comment:
I think ReadableStream may print its own console error when you declare error like this (so you could probably remove error object from log or vice versa so we don't have duplicates)

Reply:
Yes, it does. Tested by shutting and restarting ZSS to cause error

Reply:
done